request_a_govuk_domain/request/views.py

```diff
@@ -60,8 +60,14 @@ def setup(self, request, *args, **kwargs):
         # User has clicked the green button, so they're
         # starting a new journey. Therefore delete the session data
         # so that no previous answer is shown in the new journey.
-        request.session.pop(REGISTRATION_DATA, None)
-        request.session.pop(APPLICATION_REFERENCE, None)
+        if "Referer" in request.headers:
+            if "confirm" not in request.headers["Referer"]:
```

Comment:
We should always clear the session if we are in start session. I don't think we will get there by clicking back button.